src/tcgwars/logic/impl/gen7/ForbiddenLight.groovy

```diff
@@ -2158,14 +2158,18 @@ public enum ForbiddenLight implements LogicCardInfo {
               assert opp.hand
             }
             onAttack {
-              if(opp.hand.hasType(SUPPORTER)){
-                def card=opp.hand.select("Opponent's hand. Select a supporter.", cardTypeFilter(SUPPORTER)).first()
-                discard card
-                bg.deterministicCurrentThreadPlayerType=self.owner
-                bg.em().run(new PlayTrainer(card))
-                bg.clearDeterministicCurrentThreadPlayerType()
+              if (opp.hand) {
+              def randomOppHand = opp.hand.shuffledCopy()
```

Comment:
Could you indent this `if` block over by one?  Also it looks like it's missing a closing bracket